consensus/hotstuff/votecollector/combined_vote_processor.go

```diff
@@ -129,35 +136,52 @@ func (p *CombinedVoteProcessor) Process(vote *model.Vote) error {
 	}
 
 	// At this point, we have enough signatures to build a QC. Another routine
-	// might just be at this point. To avoid uplicate work, only one routine can pass:
+	// might just be at this point. To avoid duplicate work, only one routine can pass:
 	if !p.done.CAS(false, true) {
 		return nil
 	}
-	err = p.buildQC()
+
+	qc, err := p.buildQC()
 	if err != nil {
-		return fmt.Errorf("could not build QC: %w", err)
+		return fmt.Errorf("could not build QC for vote %v: %w", vote.ID(), err)
```

Comment:
A QC is built from many votes. I think tying the reason to the last vote is generally not correct. For example, the QC construction might fail, because the proposer vote (the first added vote) was invalid. 

```suggestion
		return fmt.Errorf("internal error constructing QC from votes: %w", err)
```